[Vinod] iP

Vinodjayakumar124 opened the pull request:
# Duke

Duke *frees your entire mind* of having to remember **details of tasks** to do such as 
- deadlines
- **todos**
- events

All you got to do is 
1. download it from [this page](https://github.com/Vinodjayakumar124/ip)
2. add you tasks after opening it

If you like programming, here's the main method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```
Also here is the inline version of that method `public class Main`.

> This is a famous quote about Duke

Features:
- [ ] Managing tasks
- [x] Managing deadlines
- [ ] Others

Thanks for reading! 😄

Commit:
Add Gradle support

vinod-jaya-kumar committed:
Level1: Added Greet, Echo, Exit Functionality

vinod-jaya-kumar committed:
Level 2 - Added Add, List functionality

vinod-jaya-kumar committed:
Level 3 - Added Mark as Done functionality

vinod-jaya-kumar committed:
Level 4 - Added ToDos, Events, Deadlines

vinod-jaya-kumar committed:
A-TextUiTesting : Added test cases

vinod-jaya-kumar committed:
Level-5 : Added DukeException class

vinod-jaya-kumar committed:
Level-6: Added delete functionality

vinod-jaya-kumar committed:
Added Saving, and reading from saved file functionality

vinod-jaya-kumar committed:
Added functionality to interpret dates and times

vinod-jaya-kumar committed:
Merge branch 'branch-Level-7'

vinod-jaya-kumar committed:
Merge branch 'branch-Level-8'

vinod-jaya-kumar committed:
Solve minor issues arising during merge of two branches

vinod-jaya-kumar committed:
Allow creation of new directory to store data

vinod-jaya-kumar committed:
abstracted TaskList into seperate class

vinod-jaya-kumar committed:
abstracted methods for storage class

vinod-jaya-kumar committed:
abstracted methods for UI class

vinod-jaya-kumar committed:
Abstracted methods for parser class

vinod-jaya-kumar committed:
Package all into Duke package

Vinodjayakumar124 committed:
Merge pull request #1 from nus-cs2103-AY2223S1/add-gradle-support

Add Gradle support

vinod-jaya-kumar committed:
Test ToDo and Event classes

vinod-jaya-kumar committed:
add find by substring functionality

vinod-jaya-kumar committed:
Modifications to fit coding style guide

vinod-jaya-kumar committed:
Add Javadoc to most classes

vinod-jaya-kumar committed:
Merge branch 'branch-Level-9'

vinod-jaya-kumar committed:
Merge branch 'branch-A-CodingStandard'
Resolve merge conflicts

vinod-jaya-kumar committed:
Merge branch 'branch-A-JavaDoc'
Resolve Merge Conflicts

nmtuan2001 commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,99 @@
+        return content;
+    }
+
+    public String listOfTaskForDisplay() {
```
Maybe you can add "get" to the name of this method (and the ones below) to make it more like an action?

nmtuan2001 commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,56 @@
+ */
+public class Deadline extends Task{
+    private String date = "";
+    private LocalDate dateProper;
```
Should this be properDate or something like that? Or formattedDate etc.

nmtuan2001 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,101 @@
+                    if (inputs.length <= 1) {
+                        throw new DukeException("☹ OOPS!!! The description of a deadline cannot be empty.");
+                    }
+                    String[] deadlineDescription = Parser.splitBySlash(inputString);
```
Should this be plural? Besides, is this just the description of deadline (since the second element is date)? Maybe you can change it to something else?

vinod-jaya-kumar committed:
Add GUI to Duke

vinod-jaya-kumar committed:
Merge branch 'branch-Level-10'

isanidiot commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,64 @@
+
+    /**
+     * Class Constructor using description of task, date and completion status
+     * @param taskDescription description of task
```
Javadoc comments are detailed and well written.
May I suggest leaving an empty line between description and parameter section?

isanidiot commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,211 @@
+        String[] inputs = Parser.splitBySpace(inputString);
+        String input = inputs[0];
+        String reply = "";
+        try {
```
Could you use a switch statement instead of the if...else...?

isanidiot reviewed (commented):
Neat implementation, with thoroughly written comments that really helped me understand the code!

vinod-jaya-kumar committed:
Add dialogbox colour

vinod-jaya-kumar committed:
Add ShadowJar support

vinod-jaya-kumar committed:
DialogBox: extract magic numbers into static variables

Magic numbers are used throughout code making it less readable and
understandable.

Static variables with appropriate names makes it easier to understand
the numbers and the associated meaning.

Let's remove the various numbers used inline and abstract them to a
static variable above

vinod-jaya-kumar committed:
Duke: abstract code for saving data and exception handling within
dukeReply method

Code is repeatedly used across each if-else statement within DukeReply
and doesn't follow SLAP.

Abstracting code helps improve readability and follows SLAP. Code
quality is improved with a shorter method length for dukeReply.

Let's abstract these two methods and replace existing code with them.

vinod-jaya-kumar committed:
Abstract code for getting task list size into method to follow SLAP

vinod-jaya-kumar committed:
Add javadoc to DukeException

vinod-jaya-kumar committed:
Create subpackage tasktype for Event, Deadline, ToDo and Task

All 4 files are currently in main src folder and not easily
identifiable for edits.

Moving these 4 files to separate subpackage based on the common function
allows for quick identification of files.

Let's
* Create subdirectory 'tasktype' and move all 4 files into that
subpackage
* Refactor code for TaskList to include new import
statements
* Create corresponding subpackage for Test folder for both TestEvent
and TestToDo and move these two files into it

vinod-jaya-kumar committed:
Create subdirectory view and move DialogBox and Ui files into it

vinod-jaya-kumar committed:
Create subdirectory backend and move TaskList, Storage files into it

vinod-jaya-kumar committed:
Minor corrections to files in subpackage tasktype

vinod-jaya-kumar committed:
Modified else statements in Storage to handle unexpected cases only

vinod-jaya-kumar committed:
Add JavaDoc for TaskList file

vinod-jaya-kumar committed:
Minor edits across Duke, Storage, TaskList and Ui

vinod-jaya-kumar committed:
Add assertions to code in TaskList

Vinodjayakumar124 committed:
Merge pull request #3 from Vinodjayakumar124/branch-A-CodeQuality

Merge Branch-A-CodeQuality into master

vinod-jaya-kumar committed:
Merge branch 'master' of https://github.com/Vinodjayakumar124/ip

* 'master' of https://github.com/Vinodjayakumar124/ip:
  Minor edits across Duke, Storage, TaskList and Ui
  Add JavaDoc for TaskList file
  Modified else statements in Storage to handle unexpected cases only
  Minor corrections to files in subpackage tasktype
  Create subdirectory backend and move TaskList, Storage files into it
  Create subdirectory view and move DialogBox and Ui files into it
  Create subpackage tasktype for Event, Deadline, ToDo and Task
  Add javadoc to DukeException
  Abstract code for getting task list size into method to follow SLAP
  Duke: abstract code for saving data and exception handling within dukeReply method
  DialogBox: extract magic numbers into static variables

vinod-jaya-kumar committed:
Merge branch 'master' into branch-A-Assertions

# Conflicts:
#	src/main/java/duke/TaskList.java

Vinodjayakumar124 committed:
Merge pull request #4 from Vinodjayakumar124/branch-A-Assertions

Merge Branch-Assertions with Master

vinod-jaya-kumar committed:
Merge branch 'master' of https://github.com/Vinodjayakumar124/ip

* 'master' of https://github.com/Vinodjayakumar124/ip:
  Add assertions to code in TaskList

vinod-jaya-kumar committed:
Add extension for reminder, generate list of tasks that happen in the
upcoming week

vinod-jaya-kumar committed:
Merge branch 'branch-B-Reminders'

vinod-jaya-kumar committed:
Re-add missing assert statments

vinod-jaya-kumar committed:
Improve GUI, Abstract code in Duke to follow SLAP principle

vinod-jaya-kumar committed:
Modify replies to match AmongUs personality
- missions instead of tasks

vinod-jaya-kumar committed:
Add Ui.png

vinod-jaya-kumar committed:
Update Ui.png

vinod-jaya-kumar committed:
Update User Guide
